### ISSUE TITLE
[Design System] Web Components - Q2 2022

### ISSUE THREAD

**Issue by caw310** [department-of-veterans-affairs/va.gov-team#39119]
### **Problem Statement**
In order to have a consistent user experience, individual components in applications and content should be identical across the user experience. For example, the user interface of a simple form should not appear or behave differently in different parts of VA.gov. However we currently do not maintain a library of components that is technically agnostic, so teams building in each of the three technical approaches draw from different shared/reusable code bases.

The Forms Library Team is beginning work on building out their proposed vision for a new forms library and will need to use Design System components as part of their work. We would like to provide them with web component versions of the Design System form control components as well as several other components that would be useful for them.

How might we create a technically agnostic approach to our design system and the components and patterns within it that will allow a team to build a new React app, new content, or hybrids without recreating components from a specific technical stack or creating more code that is unique to a single component.

### Hypothesis or Bet
VFS teams will be able to build more consistent applications across different technical environments

VSP will be able to maintain a single source of truth for multiple technical environments

### We will know we're done when... ("Definition of Done")
We have:
Created 4 new web components
- Maintenance Banner - removing as not priority
- Date 
- TextArea
- Number input (part of date) 
- Button - adding as it is needed by FLT
- Month / Year - adding as a variant of Date
- [stretch] System Down View - removing

[Stretch] Improve existing web components
- TextInput
- Radio
- Checkbox
- Checkbox-group

### Known Blockers/Dependencies
n/a

### Projected Launch Date
By end of Q2 2022. 
Targeting end of June 2022

### Required Artifacts
**Documentation**
PRODUCT_NAME: Web Components
Product Outline: Web Components Product Outline Web Component Product Outline
https://vfs.atlassian.net/wiki/spaces/DST/pages/2040463431/Web+Component+Product+Outline  

### Measurement
Success metrics: 
- Increased use of web components version and decrease in React components

**Comment by EricaRobbins** [comment]
@humancompanion-usds - Status Check: **Have we completed this item yet?**
I noticed the expected delivery was "by end of Q2 2022 - Targeting end of May 2022".

If not, please provide an estimated delivery date - thanks in advance!

**Comment by humancompanion-usds** [comment]
@EricaRobbins - I'm not sure where you are seeing end of May but my understanding is that the end of Q2 is the end of June. Perhaps we updated this issue and didn't update another source of info? @caw310 may know. At any rate, you can see from this Epic that we have 20 of 35 issues completed and thus we have more work to do.

**Comment by caw310** [comment]
@EricaRobbins @humancompanion-usds , teams were asked to swarm on initiatives so we were aiming to have the web components we identified for Q2 to be built by end of May. However, due to team changes and shifting priorities, we were not able to build all the web components by end of May. In addition, the web components we were going to build changed. This was due to requests from FLT and the fact that we had a developer leave the team. We should be done with the web components we identified for Q2 by end of June.
